[KubeRay][Doc] KubeRay doc migration (1/n) (#38647)
ray-project/ray: doc/source/cluster/kubernetes/index.md

  Find source document here: https://docs.google.com/drawings/d/1E3FQgWWLuj8y2zPdKXjoWKrfwgYXw6RV_FWRwK8dVlg/edit
```

KubeRay introduces three distinct Kubernetes Custom Resource Definitions (CRDs): **RayCluster**, **RayJob**, and **RayService**.

> These are designed to assist users in efficiently managing Ray clusters tailored to various use cases.

Reviewer: ```suggestion
These CRDs assist users in efficiently managing Ray clusters tailored to various use cases.
```

> Check out the [Getting Started](kuberay-quickstart) page to learn the basics of KubeRay and follow the quick start guides to run your first Ray application on Kubernetes with KubeRay!

Reviewer: ```suggestion
See [Getting Started](kuberay-quickstart) to learn the basics of KubeRay and follow the quickstart guides to run your first Ray application on Kubernetes with KubeRay.
```

            :expand:

            Try example workloads

    .. grid-item-card::

        **Ecosystem**
        ^^^
    

>         Integrate KubeRay with 3rd party Kubernetes ecosystem tools.

Reviewer: ```suggestion
        Integrate KubeRay with third party Kubernetes ecosystem tools.
```

    
        +++
        .. button-ref:: kuberay-ecosystem-integration
            :color: primary
            :outline:
            :expand:

>             Explore Kubernetes ecosystem

Reviewer: ```suggestion
            Ecosystem Guides
```

    .. grid-item-card::

        **Benchmarks**
        ^^^
    
        Check the KubeRay benchmark results.
    
        +++
        .. button-ref:: kuberay-benchmarks
            :color: primary
            :outline:
            :expand:

>             Check the benchmark results

Reviewer: ```suggestion
            Benchmark results
```

    
    .. grid-item-card::

        **Troubleshooting**
        ^^^
    

>         Check the KubeRay troubleshooting guides.

Reviewer: ```suggestion
        Consult the KubeRay troubleshooting guides.
```

    
        +++
        .. button-ref:: kuberay-troubleshooting
            :color: primary
            :outline:
            :expand:

>             Check the troubleshooting guides

Reviewer: ```suggestion
            Troubleshooting guides
```
